packages/x-scheduler-headless/src/utils/SchedulerStore/SchedulerStore.utils.ts

```diff
@@ -204,21 +218,41 @@ export function buildResourcesState<TEvent extends object, TResource extends obj
   parameters: Pick<SchedulerParameters<TEvent, TResource>, 'resources' | 'resourceModelStructure'>,
 ): Pick<
   SchedulerState<TEvent>,
-  'resourceIdList' | 'processedResourceLookup' | 'resourceModelStructure'
+  | 'resourceIdList'
+  | 'processedResourceLookup'
+  | 'resourceModelStructure'
+  | 'resourceChildrenIdLookup'
 > {
   const { resources = EMPTY_ARRAY, resourceModelStructure } = parameters;
 
   const resourceIdList: string[] = [];
   const processedResourceLookup = new Map<CalendarResourceId, CalendarResource>();
+  const resourceChildrenIdLookup = new Map<CalendarResourceId, CalendarResourceId[]>();
+
+  const addResourceToState = (processedResource: CalendarResource) => {
+    const { children, ...resourceWithoutChildren } = processedResource;
+    processedResourceLookup.set(processedResource.id, resourceWithoutChildren);
+    if (children) {
+      for (const child of children) {
+        if (!resourceChildrenIdLookup.get(processedResource.id)) {
+          resourceChildrenIdLookup.set(processedResource.id, []);
```

Comment:
I think you should be able to generate `resourceChildrenIdLookup` in a selector instead of storing it in the state. It' just derived from `processedResourceLookup` right?

Reply:
OK we can keep it as is and refacto later if needs be :+1: